dotcom-rendering/src/lib/getLiveblogAdPositions.ts

```diff
@@ -0,0 +1,48 @@
+import {
+	calculateApproximateBlockHeight,
+	shouldDisplayAd,
+} from './liveblogAdSlots';
+
+/**
+ * Decides where ads should be inserted on liveblogs/deadblogs.
+ */
+const getLiveblogAdPositions = (blocks: Block[], isMobile: boolean): number[] =>
+	blocks.reduce<{
+		heightSinceAd: number;
+		adCounter: number;
+		adPositions: number[];
+	}>(
+		(accumulator, block, index) => {
+			const { heightSinceAd, adCounter, adPositions } = accumulator;
+
+			const updatedPxSinceAd =
+				heightSinceAd +
+				calculateApproximateBlockHeight(block.elements, isMobile);
+
+			const willInsertAd = shouldDisplayAd(
+				index + 1,
+				blocks.length,
+				adCounter,
+				updatedPxSinceAd,
+				isMobile,
+			);
+
+			if (willInsertAd) {
+				adPositions.push(index);
+
+				return {
+					...accumulator,
```

Comment:
Love this use of reduce, just wondering if it's worth destructuring the accumulator the function parameter and explicitly using the variables here instead of spreading.

It took me a sec to realise where `adPositions` was going to be used 😅 (it's returned as part of `...accumulator` right?)

Reply:
Yep that's right. Sure, I'll make it explicit.